test/Constraints/bridging.swift

```diff
@@ -288,7 +288,7 @@ func rdar19836341(_ ns: NSString?, vns: NSString?) {
 // <rdar://problem/20029786> Swift compiler sometimes suggests changing "as!" to "as?!"
 func rdar20029786(_ ns: NSString?) {
   var s: String = ns ?? "str" as String as String // expected-error{{cannot convert value of type 'NSString?' to expected argument type 'String?'}}
-  var s2 = ns ?? "str" as String as String // expected-error {{binary operator '??' cannot be applied to operands of type 'NSString?' and 'String'}} expected-note{{}}
+  var s2 = ns ?? "str" as String as String // expected-error {{cannot convert value of type 'String' to expected argument type 'NSString'}}
```

Comment:
As a user I think it makes more sense to get an error about "str" as String producing unexpected type of "String" instead of "NString" which would much an "NSString?" in that case, because original error is not every helpful to figure out what is actually wrong...

Here is the output of the new diagnostic:
```
<REPL Input>:2:44: error: cannot convert value of type 'String' to expected argument type 'NSString'
            var s2 = ns ?? "str" as String as String
                           ~~~~~~~~~~~~~~~~^~~~~~~~~
                                                     as NSString
```